sycl/doc/extensions/LevelZeroBackend/LevelZeroBackend.md

```diff
@@ -58,39 +58,170 @@ and they must be included in the order shown:
+</tr><tr>
+<td>
+
+``` C++
+make_context<backend::ext_oneapi_level_zero>(
+    const backend_input_t<
+        backend::ext_oneapi_level_zero, context> &)
+```
+</td>
+<td>Constructs a SYCL context instance from a Level-Zero <code>ze_context_handle_t</code>. The context is created against the devices passed in <code>DeviceList</code> structure member. There must be at least one device given and all the devices must be from the same SYCL platform and thus from the same Level-Zero driver. The <code>Ownership</code> input structure member specifies if the SYCL runtime should take ownership of the passed native handle. The default behavior is to transfer the ownership to the SYCL runtime. See section 4.4 for details.</td>
+</tr><tr>
+<td>
+
+``` C++
+make_queue<backend::ext_oneapi_level_zero>(
+    const backend_input_t<
+        backend::ext_oneapi_level_zero, queue> &,
+    const context &Context)
+```
+</td>
+<td>Constructs a SYCL queue instance from a Level-Zero <code>ze_command_queue_handle_t</code>. The <code>Context</code> argument must be a valid SYCL context encapsulating a Level-Zero context. The queue is attached to the first device in the passed SYCL context. The <code>Ownership</code> input structure member specifies if the SYCL runtime should take ownership of the passed native handle. The default behavior is to transfer the ownership to the SYCL runtime. See section 4.4 for details.</td>
+</tr><tr>
+<td>
+
+``` C++
+make_event<backend::ext_oneapi_level_zero>(
+    const backend_input_t<
+        backend::ext_oneapi_level_zero, event> &,
+    const context &Context)
+```
+</td>
+<td>Constructs a SYCL event instance from a Level-Zero <code>ze_event_handle_t</code>. The <code>Context</code> argument must be a valid SYCL context encapsulating a Level-Zero context. The Level-Zero event should be allocated from an event pool created in the same context. The <code>Ownership</code> input structure member specifies if the SYCL runtime should take ownership of the passed native handle. The default behavior is to transfer the ownership to the SYCL runtime. See section 4.4 for details.</td>
+</tr><tr>
+<td>
+
+``` C++
+make_kernel_bundle<backend::ext_oneapi_level_zero,
+                   bundle_state::executable>(
+    const backend_input_t<
+        backend::ext_oneapi_level_zero,
+        kernel_bundle<bundle_state::executable>> &,
+    const context &Context)
+```
+</td>
+<td>Constructs a SYCL kernel_bundle instance from a Level-Zero <code>ze_module_handle_t</code>. The <code>Context</code> argument must be a valid SYCL context encapsulating a Level-Zero context. The Level-Zero module must be fully linked (i.e. not require further linking through [<code>zeModuleDynamicLink</code>](https://spec.oneapi.com/level-zero/latest/core/api.html?highlight=zemoduledynamiclink#_CPPv419zeModuleDynamicLink8uint32_tP18ze_module_handle_tP28ze_module_build_log_handle_t)), and thus the SYCL kernel_bundle is created in the "executable" state.</td>

```

Comment:
The formatting for the link on this line is incorrect because it mixes markdown with HTML.  As a result, the rendered spec look terrible here.

The following comments I made from #4542 apply here also:

* I think we need to be more precise about the requirements of the context parameter and say that the `ze_module_handle_t ` handle must have been created from this same context.

* If the default behavior is "transfer", then I think we need to add another requirement.  Since the runtime takes ownership of the native handle in this case, I think the application must also promise not to have any outstanding `ze_kernel_handle_t` handles to the underlying `ze_module_handle_t`. Otherwise, the runtime won't be able to deallocate that `ze_module_handle_t`.

If you want to address these in #4542 instead, though, that's fine with me.

Reply:
> If you want to address these in #4542 instead, though, that's fine with me.

Yes, that's what I will do.